core/trino-main/src/main/java/io/trino/sql/ir/Comparison.java

```diff
@@ -87,8 +88,7 @@ public Operator negate()
     public Comparison
     {
         requireNonNull(operator, "operator is null");
-        requireNonNull(left, "left is null");
-        requireNonNull(right, "right is null");
+        validateType(left.type(), right);
```

Comment:
These should match the operator... matching each other it nice also

Reply:
We don't have the ResolvedFunction in Comparison yet. Once I add it, we can also validate against it.